src/internal/signalr/handler.go

```diff
@@ -22,40 +23,30 @@ type Handler interface {
 }
 
 type observationsHandler struct {
-	chargepoint chargepoint.Service
-	meterElec   numericmeter.Service
-	cache       cache.Cache
-	callbacks   map[ObservationID]func(Observation) error
+	cache     cache.Cache
+	callbacks map[ObservationID]func(Observation) error
+	thing     adapter.Thing
 
 	isCloudOnline atomic.Bool
 	isStateOnline atomic.Bool
 }
 
 // NewObservationsHandler creates new observation handler.
 func NewObservationsHandler(thing adapter.Thing, cache cache.Cache) (Handler, error) {
-	chargepoint, err := getChargepointService(thing)
-	if err != nil {
-		return nil, err
-	}
-
-	meterElec, err := getMeterElecService(thing)
-	if err != nil {
-		return nil, err
-	}
-
 	handler := observationsHandler{
-		chargepoint: chargepoint,
-		meterElec:   meterElec,
-		cache:       cache,
+		cache: cache,
+		thing: thing,
 	}
 
 	handler.isCloudOnline.Store(true)
 	handler.isStateOnline.Store(true)
 
 	handler.callbacks = map[ObservationID]func(Observation) error{
+		PhaseMode:             handler.handlePhaseMode,
```

Comment:
Add support for observing a number of phases and grid type. Please add the logic to update inclusion report as well (may have an early exit if the state of the service did not change).